evalml/tests/component_tests/test_oversampler.py

```diff
@@ -447,14 +450,117 @@ def test_oversampler_handle_nullable_types(
-    X, y = oversampler._handle_nullable_types(X, y)
-    oversampler.fit(X, y)
-    X_t, y_t = oversampler.transform(X, y)
+    oversampler.fit(X.ww.copy(), y)
+    X_t, y_t = oversampler.transform(X.ww.copy(), y)
 
     # Confirm oversampling happened by checking the length increased
     assert len(X_t) > len(X)
     assert len(y_t) > len(y)
+
+    # Confirm the original types are maintained
+    assert original_schema == X_t.ww.schema
+
+
+@patch(
+    "evalml.pipelines.components.component_base.ComponentBase._handle_nullable_types",
+)
+def test_oversampler_calls_handle_nullable_types(
+    mock_handle_nullable_types,
+    X_y_binary,
+):
+    X, y = X_y_binary
+
+    oversampler = Oversampler()
+
+    assert not mock_handle_nullable_types.called
+    mock_handle_nullable_types.return_value = X, y
+
+    oversampler.fit(X, y)
+    assert not mock_handle_nullable_types.called
+
+    oversampler.transform(X, y)
+    assert mock_handle_nullable_types.called
+
+
+@pytest.mark.parametrize(
+    "nullable_y_ltype",
+    ["IntegerNullable", "AgeNullable", "BooleanNullable"],
+)
+@pytest.mark.parametrize(
+    "im_oversampler",
+    [
+        im.SMOTENC(categorical_features=[0]),
+        im.SMOTEN(),
+    ],
+)
+@pytest.mark.parametrize(
+    "handle_incompatibility",
+    [
+        True,
+        pytest.param(
+            False,
+            marks=pytest.mark.xfail(strict=True, raises=ValueError),
+        ),
+    ],
+)
+def test_oversampler_nullable_type_incompatibility(
+    nullable_type_test_data,
+    handle_incompatibility,
+    im_oversampler,
+    nullable_y_ltype,
+):
+    """Testing that the nullable type incompatibility that caused us to add handling for Oversampler
+    is still present in imblearn's SMOTE oversamplers. If this test is causing the test suite to fail
+    because the code below no longer raises the expected ValueError, we should confirm that the nullable
+    types now work for our use case and remove the nullable type handling logic from Oversampler.
+    """
+    X = nullable_type_test_data(has_nans=False)
+    X = X.ww[
+        ["categorical col", "int col nullable", "bool col nullable", "age col nullable"]
+    ]
+    y = ww.init_series(pd.Series([1, 0, 0, 1, 1] * 4), logical_type=nullable_y_ltype)
+
+    if handle_incompatibility:
+        evalml_oversampler = Oversampler(sampling_ratio=0.5)
+        X, y = evalml_oversampler._handle_nullable_types(X, y)
+
+    # We have to convert to `object` dtype with categorical columns
+    # because of the sklearn bug described by the test below this one
+    X["categorical col"] = X["categorical col"].astype("object")
+
+    im_oversampler.fit_resample(X, y)
+
+
+@pytest.mark.parametrize(
+    "handle_incompatibility",
+    [
+        True,
+        pytest.param(
+            False,
+            marks=pytest.mark.xfail(strict=True, raises=ValueError),
+        ),
+    ],
+)
+def test_oversampler_category_dtype_incompatibility(
```

Comment:
This is actually the same bug as the nullable type incompatibility, and it's why we convert categorical columns to "object" in [base_sampler](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/components/transformers/samplers/base_sampler.py#L99-L105). There is a fix in the works https://github.com/scikit-learn/scikit-learn/pull/25814 (because of the wonderful @thomasjpfan who has been a key contributor to our ability to support nullable types so widely), so I wanted to add this xfailing test to remind us to also remove that logic at the same time.

Reply:
@eccabay converting `caegory` to `object` was a bug fix of yours from a [while back](https://github.com/alteryx/evalml/pull/3732), so I just want to make sure you get a chance to take a look at this test in case I'm missing something here.